src/vs/workbench/services/languageRuntime/common/languageRuntimeUiClient.ts

```diff
@@ -161,9 +161,14 @@ export class UiClientInstance extends Disposable {
 
 		// Wrap the ShowHtmlFile event to start a proxy server for the HTML file.
 		this._register(this._comm.onDidShowHtmlFile(async e => {
+			let uri: URI;
 			try {
-				// Start an HTML proxy server for the file
-				const uri = await this.startHtmlProxyServer(e.path);
+				if (e.path.endsWith('.html') || e.path.endsWith('.htm')) {
+					// Start an HTML proxy server for the file
+					uri = await this.startHtmlProxyServer(e.path);
+				} else {
+					uri = URI.parse(e.path);
+				}
```

Comment:
Does it work without this change? This looks like it won't create a proxy server which I think wouldn't work on Workbench.

Reply:
This part I wasn't sure if it would work. When opening other URLs in the viewer, it doesn't proxy it. I'll try it out but I agree that I don't think it would work.

Reply:
It seems to work when running Positron server but it's all local. I think it's only working since the deepest iframe loads the plotly url.

@sharon-wang I think it probably needs a tweak on creating the proxy server or a slightly different type of proxy server. The one used for HTML files didn't seem to work with the localhost URL that plotly produces (file URI vs. localhost URL).

Reply:
Do you have some example plotly URI/URLs? On `main`, python plotly plots work on Server Web and Workbench, so I wonder what changes to the URI/URLs are occurring?

Reply:
`http://127.0.0.1:6728/`

It's loading something like that with a random port. Before this change, it would load an HTML file and go through the `HTMLProxy`. So I think it'll need to load the URL with a proxy as it did before.

Reply:
@timtmok could you please verify if reverting this change to bypass the proxy and bringing in the changes from https://github.com/posit-dev/positron/pull/4855 still works with this PR?

Reply:
It does get proxied successfully when running it locally (in web and desktop). I'm still setting up Workbench to see how it behaves there but it's looking promising.